src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVIntervalTree.java

```diff
@@ -0,0 +1,904 @@
+        }
+
+        @Override
+        public V1 setValue( final V1 value ) {
+            final V1 result = this.value;
+            this.value = value;
+            return result;
+        }
+
+        int getSize() {
+            return size;
+        }
+
+        SVInterval getMaxEndInterval() {
+            return maxEndInterval;
+        }
+
+        Node<V1> getLeft() {
+            return left;
+        }
+
+        Node<V1> insertLeft( final SVInterval interval, final V1 value, final Node<V1> root ) {
+            left = new Node<>(this, interval, value);
+            return insertFixup(left, root);
+        }
+
+        Node<V1> getRight() {
+            return right;
+        }
+
+        Node<V1> insertRight( final SVInterval interval, final V1 value, final Node<V1> root ) {
+            right = new Node<>(this, interval, value);
+            return insertFixup(right, root);
+        }
+
+        Node<V1> getNext() {
+            Node<V1> result;
+
+            if ( right != null ) {
+                result = right;
+                while ( result.left != null ) {
+                    result = result.left;
+                }
+            } else {
+                Node<V1> node = this;
+                result = parent;
+                while ( result != null && node == result.right ) {
+                    node = result;
+                    result = result.parent;
+                }
+            }
+
+            return result;
+        }
+
+        Node<V1> getPrev() {
+            Node<V1> result;
+
+            if ( left != null ) {
+                result = left;
+                while ( result.right != null ) {
+                    result = result.right;
+                }
+            } else {
+                Node<V1> node = this;
+                result = parent;
+                while ( result != null && node == result.left ) {
+                    node = result;
+                    result = result.parent;
+                }
+            }
+
+            return result;
+        }
+
+        boolean wasRemoved() {
+            return size == 0;
+        }
+
+        Node<V1> remove( final Node<V1> initialRoot ) {
+            Node<V1> root = initialRoot;
+            if ( size == 0 ) {
+                throw new IllegalStateException("Entry was already removed.");
+            }
+
+            if ( left == null ) {
+                if ( right == null ) { // no children
+                    if ( parent == null ) {
+                        root = null;
+                    } else if ( parent.left == this ) {
+                        parent.left = null;
+                        fixup(parent);
+
+                        if ( isBlack )
+                            root = removeFixup(parent, null, root);
+                    } else {
+                        parent.right = null;
+                        fixup(parent);
+
+                        if ( isBlack )
+                            root = removeFixup(parent, null, root);
```

Comment:
This one needs braces.  We've had to many easy to miss bugs from someone coming in and adding another line in the indented block and forgetting to add braces.